Dockerfile

```diff
@@ -0,0 +1,19 @@
+ARG BUILD_TAG=8-jdk-alpine
+ARG RUN_TAG=8-jre-alpine
+
+FROM openjdk:${BUILD_TAG} AS build-env
+WORKDIR /src
+
+# Copy src .dockerignore prevents copy of unnecessary files
+COPY . .
+
+# Gradle build with daemon disabled
+RUN ./gradlew build --no-daemon --stacktrace
```

Comment:
What do we gain by doing the build inside docker?

Reply:
We remove the need for the build machine to have the necessary build tools. I believe it to be more portable.